From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 10 Apr 2023 16:06:19 +0200
Subject: tests: allocators: Fix fdmem test

Since the test wants to close the FD itself, we need to tell the FD
allocator not to. Otherwise the test will fail with recent GLib, which
emits a critical warning on EBADF.

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4381
---
 tests/check/libs/allocators.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tests/check/libs/allocators.c b/tests/check/libs/allocators.c
index 6b974b2..da08ae0 100644
--- a/tests/check/libs/allocators.c
+++ b/tests/check/libs/allocators.c
@@ -80,7 +80,8 @@ GST_START_TEST (test_fdmem)
 
   alloc = gst_fd_allocator_new ();
   fail_unless (alloc);
-  mem = gst_fd_allocator_alloc (alloc, fd, 10, GST_FD_MEMORY_FLAG_KEEP_MAPPED);
+  mem = gst_fd_allocator_alloc (alloc, fd, 10,
+      GST_FD_MEMORY_FLAG_KEEP_MAPPED | GST_FD_MEMORY_FLAG_DONT_CLOSE);
 
   fail_unless (gst_memory_map (mem, &info, GST_MAP_READ));
   fail_unless (info.data[5] == '5');
@@ -95,7 +96,7 @@ GST_START_TEST (test_fdmem)
   gst_memory_unmap (mem, &info);
 
   gst_memory_unref (mem);
-  fail_unless (g_close (fd, NULL) == 0);
+  fail_unless (g_close (fd, NULL));
   gst_object_unref (alloc);
 }
 
